drivers/gpu/drm/xen/xen_drm_zcopy_drv.c

```diff
@@ -133,23 +132,19 @@ static inline void xen_wait_obj_free(struct xen_drv_info *drv_info,
 	spin_unlock(&drv_info->wait_list_lock);
 }
 
-static void xen_wait_obj_free_all(struct xen_drv_info *drv_info)
+static inline void xen_wait_obj_check_pending(struct xen_drv_info *drv_info)
 {
-	struct xen_wait_obj *wait_obj, *q;
-
-	spin_lock(&drv_info->wait_list_lock);
-	list_for_each_entry_safe(wait_obj, q, &drv_info->wait_obj_list, list) {
-		complete_all(&wait_obj->completion);
-		list_del(&wait_obj->list);
-		kfree(wait_obj);
+	if (!list_empty(&drv_info->wait_obj_list)) {
```

Comment:
`BUG_ON(!list_empty(&drv_info->wait_obj_list)`

You can add comment, describing why this is bad.